src/lib/node-oauth-client-provider.ts

```diff
@@ -58,6 +64,13 @@ export class NodeOAuthClientProvider implements OAuthClientProvider {
    */
   async clientInformation(): Promise<OAuthClientInformationFull | undefined> {
     // log('Reading client info')
+    if (this.clientId) {
```

Comment:
much cleaner if this is just `if (this.staticClientInformation) return this.staticClientInformation` here